library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -1110,6 +1116,16 @@ public String toString()
       return "DirectoryScanner [filePatternRegexp=" + filePatternRegexp + " partitionIndex=" +
           partitionIndex + " partitionCount=" + partitionCount + "]";
     }
+
+    protected void setPartitionIndex(int partitionIndex)
+    {
+      this.partitionIndex = partitionIndex;
+    }
+
+    protected void setPartitionCount(int partitionCount)
+    {
+      this.partitionCount = partitionCount;
+    }
```

Comment:
Don't see a test for the newly added methods setPartitionIndex and setPartitionCount (since the rationale in the ticket is "Inherited DirectoryScanner can not set them", we should have a test that shows that this situation has
been remedied).